pkg/adapter/cronjobevents/adapter.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cronjobevents
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/robfig/cron"
+
+	"github.com/knative/pkg/cloudevents"
+	"github.com/knative/pkg/logging"
+	"github.com/knative/pkg/signals"
+	"go.uber.org/zap"
+)
+
+const (
+	eventType = "dev.knative.cronjob.event"
+)
+
+// Adapter implements the Cron Job adapter to trigger a Sink.
+type Adapter struct {
+	// Schedule is a cron format string such as 0 * * * * or @hourly
+	Schedule string
+
+	// Data is the data to be posted to the target.
+	Data string
+
+	// SinkURI is the URI messages will be forwarded on to.
+	SinkURI string
+}
+
+func (a *Adapter) Start(ctx context.Context) error {
+	logger := logging.FromContext(ctx)
+
+	sched, err := cron.ParseStandard(a.Schedule)
+	if err != nil {
+		logger.Error("Unparseable schedule: ", a.Schedule, zap.Error(err))
+		return err
+	}
+	c := cron.New()
+	stopCh := signals.SetupSignalHandler()
+	cctx, cancel := context.WithCancel(ctx)
+	c.Schedule(sched, cron.FuncJob(func() {
+		m := Message{ID: a.generateMessageId(), EventTime: time.Now().UTC(), Body: a.Data}
+		err = a.postMessage(cctx, &m)
+		if err != nil {
+			logger.Error("Failed to post message.", zap.Error(err))
+			return
+		}
+		logger.Debug("Finished posting message job.", zap.Any("message", m))
+	}))
+	go c.Start()
```

Comment:
Don't need to do this in a goroutine, [Start](https://github.com/robfig/cron/blob/master/cron.go#L138) does that automatically.